user/user.go

```diff
@@ -452,9 +461,28 @@ func ParseUpdateUserDetails(reader io.Reader) (*UpdateUserDetails, error) {
 	details := &UpdateUserDetails{}
 	if err := details.ExtractFromJSON(reader); err != nil {
 		return nil, err
-	} else {
-		return details, nil
 	}
+
+	if details.EmailVerified != nil {
+		details.nFields += 1
+	}
+	if len(details.Emails) > 0 {
+		details.nFields += 1
+	}
+	if details.Password != nil {
+		details.nFields += 1
+	}
```

Comment:
why don't we consider the currentPassword ?

Reply:
Because it is mandatory only if we change the password.
A change with only the current password has no action associated.